src/libraries/System.Private.CoreLib/src/System/Reflection/Nullability/NullabilityInfoContext.cs

```diff
@@ -0,0 +1,482 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Reflection
+{
+    /// <summary>
+    /// Provides APIs for populating nullability information/context from reflection members:
+    /// <see cref="ParameterInfo"/>, <see cref="FieldInfo"/>, <see cref="PropertyInfo"/> and <see cref="EventInfo"/>.
+    /// </summary>
+    public sealed class NullabilityInfoContext
+    {
+        private const string CompilerServicesNameSpace = "System.Runtime.CompilerServices";
+        private readonly Dictionary<Module, NotAnnotatedStatus> _publicOnlyModules = new();
+        private readonly Dictionary<MemberInfo, NullabilityState> _context = new();
+        private static readonly NullabilityInfo[] s_emptyArray = Array.Empty<NullabilityInfo>();
```

Comment:
I believe just re-using `Array.Empty<NullabilityInfo>()` is sufficient (instead of caching).

Reply:
OK, it feels like creating and empty array

Reply:
Due to generics + static fields, the underlying array is cached per `<T>`